Automattic/wp-calypso
# Newsletter Settings: Add subscriptions module activation banner to Newsletter settings

Related to https://github.com/Automattic/wp-calypso/pull/74209
Related to https://github.com/Automattic/jetpack/pull/33065

## Proposed Changes

![eQss2G.png](https://github.com/Automattic/wp-calypso/assets/2019970/e0c0ee34-b111-4a9b-9a43-6392ac7dee38)
* For Jetpack-connected sites, show the subscription module activation banner at the top of the Newsletter settings page, but only if the subscriptions module is not active.
* Extract the `BannerActivation` component from `my-sites/people/subscribers` to `calypso/blocks` and rename it to `SubscriptionsModuleBanner`
* Reuse the `SubscriptionsModuleBanner` in the Newsletter settings page

## Testing Instructions



Test the banner for jetpack connected site in Calypso
1. Disable the subscriptions module. You can do so from the **Settings —> Discussions** in Calypso, once you switched to your jetpack connected test site.
   ![Bx5AuZ.png](https://github.com/Automattic/wp-calypso/assets/2019970/58166807-a312-40c9-a072-23fb834de9a5)
2. Go to Settings --> Newsletter 
3. Ensure the subscription module activation banner is displayed, and that the settings form appears disabled.
4. Enable the subscriptions module by clicking the **Enable** button.
5. Once the action is complete successfully the settings form should become enabled.
6. You can double check that the module was actually activated by going back to **Settings —> Discussion** and checking that the module's toggle is on. 

Based on my understanding, testing for regressions at https://wordpress.com/people/subscribers/$siteSlug isn't necessary since it has been superseded by https://wordpress.com/subscribers/$siteSlug. On a related note, it appears the directory [client/my-sites/people/subscribers](https://github.com/Automattic/wp-calypso/tree/7504068fdc6ecc0efe8ff6fca87c9a6421e45c6f/client/my-sites/people/subscribers) and [client/my-sites/people/subscriber-details](https://github.com/Automattic/wp-calypso/tree/7504068fdc6ecc0efe8ff6fca87c9a6421e45c6f/client/my-sites/people/subscriber-details) might be redundant and could be cleaned up.
cc @yansern @bogiii 

## Pre-merge Checklist
